express-notes-api/write-file.js

```diff
@@ -0,0 +1,13 @@
+const fs = require('fs');
+const data = require('./data.json');
+
+const writeFile = () => {
+  fs.writeFile('data.json', JSON.stringify(data, null, 2), err => {
+    if (err) {
+      console.error(err);
+      process.exit(1);
+    }
+  });
+};
+
+module.exports = writeFile;
```

Comment:
I like that you made a function to write the data and even put it in its own file, however this won't work as expected. You would always be writing the same data back to itself, not the modified data from the other file. Also since this is async you can't properly catch errors or send a response once the wrote process has completed. Also in a server you don't want to exit the process if there is an error, this will stop your server completely making it impossible to handle any more requests. Not a good plan for a web server. You can refactor this a bit to make it work and more versatile.

```JAVASCRIPT
const fs = require('fs');

const writeFile = (data, callback) => {
  fs.writeFile('data.json', JSON.stringify(data, null, 2), callback);
};

module.exports = writeFile;
```